projects/pureth-eip-7919-erigon.md

```diff
@@ -0,0 +1,173 @@
+# Pureth Proposal for Erigon
+
+## Introduction
+
+Ethereum clients currently speed up `eth_getLogs` queries using bloom filters, but these filters can generate many false positives—especially as event density increases. To address this, Zsolt introduced **filtermaps**, now implemented in Geth. This proposal details a plan to implement filtermaps in **Erigon**, leveraging EIP-7745 for correctness, completeness and scalability.
+
+## Motivation
+
+False positive ratio in bloom filter rises significantly as the number of events per filter . We want a system that is Correct , Complete and Scalable so EIP 7745 was introduced along with filtermaps 
+
+## Importance
+
+- **Reduces False Positives:** Filtermaps minimize unnecessary log scans .
+- **Efficient Log Queries:** Servers avoid sending entire index ranges, optimizing network and compute.
+- **Performance:** Benchmarks show notable speed ([see results](https://xcancel.com/sina_mahmoodi/status/1930565142183387627)).
+
+## Project Description
+
+**Goal:**  
+Implement filtermaps in Erigon, closely following Zsolt's design in Go-Ethereum (Geth).
+
+### Key Steps
+
+1. **Define Data Structures:** Adapt struct and field definitions for Erigon compatibility.
+2. **Design Interfaces:** Create matcher, indexer, and filtermap interfaces for modular implementation.
+3. **Implement Core Components:** Develop matcher logic for log value queries, indexer for event indexing, and filtermaps for efficient storage and retrieval.
```

Comment:
One step that is missing here is to distinguish the accelerating data structure (the filter maps), which can be done independently without a hard fork, and works even on mainnet, as demonstrated by Geth in https://xcancel.com/sina_mahmoodi/status/1930565142183387627; from the hashing of the log index and putting the root hash into the block header, which requires a hard fork but is crucial to enable verifiable endpoints (as the proof is compared against the block header).

So, suggesting to add an additional step to put the root hash into the block header.